### DISCUSSION TITLE
Command line argument or URI to open file and jump to a specific node ID?

### DISCUSSION THREAD

**Discussion by hi2u** [freeplane/freeplane#835] [Help] [answered]
Let's say I have a node in `myfile.mm` with ID `ID_31337`:

```xml
<node TEXT="The text shown on the node" ID="ID_31337" CREATED="1662856168616" MODIFIED="1662856177991"/>
```

From outside of Freeplane itself, I'd like a command to run that will launch Freeplane with `myfile.mm`, and then immediately jump to + focus that node.  e.g. Something like:

    freeplane.exe --goto="ID_31337" myfile.mm

...or if the file is already open, just navigate to it in the pre-existing window.

Is there any way to do this on the command line?

Or otherwise something like a URI protocol handler that could do this?  Something like:

    freeplane:///c:/directory-path/myfile.mm?id=ID_31337

Basically I want to externally index all the nodes in my mindmaps in some other software, and then be able to launch + navigate to them directly from outside of the Freeplane GUI.

**Comment by steveatmaat** [comment, marked as answer]
Select a node.
From the menu, Edit>Copy>Copy node URI. (Also available on the right-click menu.)

I'm on Windows, and the URI works fine from the Windows command line or used as a link in another app.

**Reply by hi2u** [reply]
Awesome, thanks so much!
